tests/ui/coherence/coherence-fn-covariant-bound-vs-static.rs

```diff
@@ -1,10 +1,15 @@
-// Test that impls for these two types are considered ovelapping:
+//@ check-pass
+
+// These types were previously considered equal as they are subtypes of each other.
+// This has been changed in #118247 and we now consider them to be disjoint.
+//
+// In our test:
 //
 // * `for<'r> fn(fn(&'r u32))`
 // * `fn(fn(&'a u32)` where `'a` is free
 //
-// This is because, for `'a = 'static`, the two types overlap.
-// Effectively for them to be equal to you get:
+// These were considered equal because, for `'a = 'static`: we check subtyping in both directions
+// and in our case.
```

Comment:
this wording feels off :thinking: 

```suggestion
// These were considered equal as for `'a = 'static` subtyping succeeds in both
// directions:
```